src/cmd/oui.rs

```diff
@@ -138,6 +223,74 @@ impl Submit {
     }
 }
 
+impl Update {
+    pub fn run(&self, opts: Opts) -> Result {
+        let password = get_password(false)?;
+        let wallet = load_wallet(opts.files)?;
+        let keypair = wallet.decrypt(password.as_bytes())?;
+        let api_client = Client::new_with_base_url(api_url(wallet.public_key.network));
+
+        let (oui, commit, nonce, update) = match self {
+            Update::Routers(routers) => (
+                routers.oui,
+                routers.commit,
+                routers.nonce,
+                helium_api::blockchain_txn_routing_v1::Update::UpdateRouters(UpdateRouters {
+                    router_addresses: map_addresses(routers.addresses.clone(), |v| v.to_vec())?,
+                }),
+            ),
+            Update::NewXor(filter) => (
+                filter.oui,
+                filter.commit,
+                filter.nonce,
+                helium_api::blockchain_txn_routing_v1::Update::NewXor(base64::decode(
+                    &filter.filter,
+                )?),
+            ),
+            Update::UpdateXor(update) => (
+                update.oui,
+                update.commit,
+                update.nonce,
+                helium_api::blockchain_txn_routing_v1::Update::UpdateXor(UpdateXor {
+                    index: update.index,
+                    filter: base64::decode(&update.filter)?,
+                }),
+            ),
+            Update::RequestSubset(size) => (
+                size.oui,
+                size.commit,
+                size.nonce,
+                helium_api::blockchain_txn_routing_v1::Update::RequestSubnet(size.size),
+            ),
+        };
+
+        let mut txn = BlockchainTxnRoutingV1 {
+            oui,
+            owner: keypair.public_key().into(),
+            fee: 0,
+            signature: vec![],
+            staking_fee: 0,
+            update: Some(update),
+            nonce: if let Some(nonce) = nonce {
+                nonce
+            } else {
+                //TODO: fetch nonce from API
```

Comment:
I was going to ask what the nonce field was doing in the commands, so this is why? 

You can get the last/"current" oui txn from `/v1/ouis/last` which would give you the actual oui value to use

Reply:
This is something different, I think. There's some oui_nonce in the ledger somewhere that gets incremented when an OUI updates. It's not exposed by the API yet afaik.